node/commons/compatibility-layer.ts

```diff
@@ -14,27 +10,35 @@ export const convertBiggyProduct = (
   const categories: string[] = product.categories
     ? product.categories.map((_: any, index: number) => {
         const subArray = product.categories.slice(0, index)
+
         return `/${subArray.join('/')}/`
       })
     : []
 
-
   const categoriesIds: string[] = product.categoryIds
-  ? product.categoryIds.map((_: any, index: number) => {
-      const subArray = product.categoryIds.slice(0, index + 1)
-      return `/${subArray.join('/')}/`
-    }).reverse()
-  : []
+    ? product.categoryIds
+        .map((_: any, index: number) => {
+          const subArray = product.categoryIds.slice(0, index + 1)
+
+          return `/${subArray.join('/')}/`
+        })
+        .reverse()
+    : []
 
   const skus: SearchItem[] = (product.skus || []).map(
     convertSKU(product, indexingType, tradePolicy)
   )
 
-  const allSpecifications = product.productSpecifications.concat(getSKUSpecifications(product))
+  const productSpecifications = product.productSpecifications ?? []
```

Comment:
This was required as a workaround on some products having `productSpecifications` set as `undefined`, @LucasTornai is taking a look 😄

Reply:
Just for the record, seems that it isn't affecting any other store for now, so I won't publish a fix beforehand and let this PR fix it.